ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/CombinedLedgerStateWriteOperations.scala

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.validator
+
+import com.daml.ledger.participant.state.kvutils.Raw
+
+import scala.concurrent.{ExecutionContext, Future}
+
+final class CombinedLedgerStateWriteOperations[ALogResult, BLogResult, LogResult](
```

Comment:
I have a feeling we won't use much this abstraction beyond this specific need, but I am OK with it.

Reply:
I agree, but I prefer to generalize when possible because there's fewer things you can get wrong.